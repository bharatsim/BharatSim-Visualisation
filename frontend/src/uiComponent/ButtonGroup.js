import React from 'react';

import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import { ChildrenPropTypes } from '../commanPropTypes';

const useStyles = makeStyles((theme) => ({
  buttonGroup: {
    '& > *': {
      marginLeft: theme.spacing(1),
    },
    '& > :first-child': {
      marginLeft: theme.spacing(0),
    },
  },
}));

function ButtonGroup({ children }) {
  const classes = useStyles();
  return <Box className={classes.buttonGroup}>{children}</Box>;
}

ButtonGroup.propTypes = {
  children: ChildrenPropTypes.isRequired,
};

export default ButtonGroup;
