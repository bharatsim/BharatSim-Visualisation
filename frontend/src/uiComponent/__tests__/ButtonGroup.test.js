import React from 'react';
import { render } from '@testing-library/react';
import ButtonGroup from '../ButtonGroup';

describe('<Buttongroup />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ButtonGroup>
        <button type="button">click here</button>
        <button type="button">click here</button>
      </ButtonGroup>,
    );

    expect(container).toMatchSnapshot();
  });
});
