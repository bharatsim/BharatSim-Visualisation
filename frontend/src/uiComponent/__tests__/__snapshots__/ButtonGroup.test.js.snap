// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttongroup /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-buttonGroup-1"
  >
    <button
      type="button"
    >
      click here
    </button>
    <button
      type="button"
    >
      click here
    </button>
  </div>
</div>
`;
